app/src/main/java/tech/pauly/findapet/shared/MapWrapper.kt

```diff
@@ -0,0 +1,75 @@
+package tech.pauly.findapet.shared
+
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.OnLifecycleEvent
+import android.content.Context
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.LatLngBounds
+import com.google.android.gms.maps.model.MapStyleOptions
+import com.google.android.gms.maps.model.MarkerOptions
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import tech.pauly.findapet.R
+import tech.pauly.findapet.dependencyinjection.ForApplication
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+open class MapWrapper @Inject constructor(@ForApplication val context: Context) : OnMapReadyCallback, LifecycleObserver {
+
+    private lateinit var mapView: MapView
+    private lateinit var map: GoogleMap
+
+    private val mapReadySubject: BehaviorSubject<Boolean> = BehaviorSubject.create()
+    open val mapReadyObservable: Observable<Boolean>
+        get() = mapReadySubject.delay(100, TimeUnit.MILLISECONDS)
```

Comment:
Adding a delay to this because zooming the camera around immediately after the load isn't working for me. First time using Google Maps, so maybe it's a fluke, but this isn't noticeable so I'm fine with it for now.